runtime/str.go

```diff
@@ -157,6 +157,29 @@ func strAdd(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return NewStr(stringV + stringW).ToObject(), nil
 }
 
+func strCapitalize(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if raised := checkMethodArgs(f, "capitalize", args, StrType); raised != nil {
+		return nil, raised
+	}
+	s := toStrUnsafe(args[0]).Value()
+	buf := bytes.Buffer{}
+	runes := []rune(s)
+	numRunes := len(runes)
+	writeRune := func(r rune, transform func(r rune) rune) {
+		if r <= unicode.MaxASCII {
+			r = transform(r)
+		}
+		buf.WriteRune(r)
+	}
+	if numRunes > 0 {
+		writeRune(runes[0], func(r rune) rune { return unicode.ToUpper(r) })
+	}
+	for i := 1; i < numRunes; i++ {
```

Comment:
For str I think you want to iterate over bytes, not runes. IIUC []rune(s) will treat s as a UTF-8 encoded string and convert it to 32-bit runes. So multi-byte characters will be treated as single runes and capitalized. This is not the behavior in CPython which seems to just capitalize with respect to ASCII:

```
>>> locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
'en_US.UTF-8'
>>> print u'вол'.capitalize()
Вол
>>> u'вол'.encode('utf8')
'\xd0\xb2\xd0\xbe\xd0\xbb'
>>> '\xd0\xb2\xd0\xbe\xd0\xbb'.capitalize()
'\xd0\xb2\xd0\xbe\xd0\xbb'```